ibc-apps/ics20-transfer/types/src/denom.rs

```diff
@@ -294,18 +301,34 @@ impl FromStr for PrefixedDenom {
     type Err = TokenTransferError;
 
     fn from_str(s: &str) -> Result<Self, Self::Err> {
```

Comment:
It would be helpful to add a description of the logic of this `from_str` implementation, as it's a little bit convoluted. Here's a suggestion:
```
/// Initializes a `PrefixedDenom` from a string that adheres to the format
/// `{port-id-1/channel-id-1}/{port-id-2/channel-id-2}/.../{port-id-n/channel-id-n}/base-denom`.
/// A `PrefixedDenom` exhibits between 1..n number of `{port-id/channel-id}` pairs.
/// The set of these pairs makes up the `TracePath`.
///
/// This `from_str` implementation peeks the first two segments split on the `/`
/// delimiter and attempts to convert the first segment into a `PortId` and the 
/// second segment into a `ChannelId`. This continues in a loop until a 
/// `{port-id/channel-id}` pair cannot be created from the top two segments. 
/// The remaining parts of the string are then considered the `BaseDenom`.
///
/// For example, given the following denom trace:
/// "transfer/channel-75/factory/stars16da2uus9zrsy83h23ur42v3lglg5rmyrpqnju4/dust",
/// the first two `/`-delimited segments are `"transfer"` and `"channel-75"`. The
/// first is a valid `PortId` and the second is a valid `ChannelId`, so that becomes
/// the first `{port-id/channel-id}` pair that gets added as part of the `TracePath`
/// of the `PrefixedDenom`. The next two segments are `"factory"`, which is not a
/// valid `PortId`, and `"stars16da2uus9zrsy83h23ur42v3lglg5rmyrpqnju4"`, which is
/// not a valid `ChannelId`. The loop breaks at this point, resulting in a
/// `TracePath` of `"transfer/channel-75"` and a `BaseDenom` of
/// `"factory/stars16da2uus9zrsy83h23ur42v3lglg5rmyrpqnju4/dust"`.
```

Feel free to edit the above to make the description more clear.

Reply:
I agree with Sean. The above description helps a lot. Great. 

>  The next two segments are `"factory"`, which is not a valid `PortId`

Just a small note: "factory" is a valid `PortId`.

Reply:
9d246d43